### PR TITLE
Nutrition calculator test

### PR DESCRIPTION
**Pull request by DerekY123**
Test for `NutritionCalculator.accumulateNutritionValue` and fixed bug in the method.

### CODE REVIEW

**Comment by DerekY123** [comment]
@olioDuan

File: NutritionManagement/src/calculation/NutritionCalculator.java [hunk 1]
```diff
@@ -13,25 +11,28 @@ public class NutritionCalculator {
      * @return a new NutritionValue object representing the total Nutrition Value
      */
     public static NutritionValue accumulateNutritionValue(ArrayList<NutritionValue> nutritionValues) {
-        Stream<NutritionValue> nutritions = nutritionValues.stream();
```
Comment (olioDuan) [inline NutritionManagement/src/calculation/NutritionCalculator.java:16]: A little tips: this change might slightly impact performance if the ArrayList is large, as each mapToDouble operation will create a new stream.

Improved readability.

File: NutritionManagement/src/calculation/NutritionCalculator.java [hunk 2]
```diff
@@ -13,25 +11,28 @@ public class NutritionCalculator {
...
+                nutritionValues.stream().mapToDouble(item -> item.getCarbs()).sum(),
+                nutritionValues.stream().mapToDouble(item -> item.getFat()).sum());
     }
 
     /**
      * 
      * @param days
-     * @return a new NutritionValue object representing the average Nutrition Value per day
+     * @return a new NutritionValue object representing the average Nutrition Value
+     *         per day
```
Comment (olioDuan) [inline NutritionManagement/src/calculation/NutritionCalculator.java:25]: ok for the newline :)

File: NutritionManagement/src/calculation/NutritionCalculator.java [hunk 3]
```diff
@@ -13,25 +11,28 @@ public class NutritionCalculator {
...
      */
     public static NutritionValue computeAverage(ArrayList<Day> days) {
-        Stream<NutritionValue> nutritions = days.stream().map(day -> day.getTotalNutrition());
         return new NutritionValue(
-                nutritions.mapToDouble(item -> item.getCalories()).average().orElse(0),
-                nutritions.mapToDouble(item -> item.getProtein()).average().orElse(0),
-                nutritions.mapToDouble(item -> item.getCarbs()).average().orElse(0),
-                nutritions.mapToDouble(item -> item.getFat()).average().orElse(0));
+                days.stream().map(day -> day.getTotalNutrition()).mapToDouble(item -> item.getCalories()).average()
+                        .orElse(0),
```
Comment (olioDuan) [inline NutritionManagement/src/calculation/NutritionCalculator.java:30]: Similar to `accumulateNutritionValue`, improves consistency across methods.

`average().orElse(0)` is a good practice, as it provides a default value in cases where the stream is empty, preventing potential _NoSuchElementException._

File: NutritionManagement/src/test/NutritionCalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import static org.junit.Assert.assertEquals;
+import java.util.ArrayList;
+import java.lang.Math;
+
+import org.junit.Test;
+
+import main.NutritionValue;
+import calculation.NutritionCalculator;
+
+public class NutritionCalculatorTest {
```
Comment (olioDuan) [inline NutritionManagement/src/test/NutritionCalculatorTest.java:12]: well-structured

File: NutritionManagement/src/test/NutritionCalculatorTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    @Test
+    public void accumulateNutritionValueEmpty() {
+        ArrayList<NutritionValue> input = new ArrayList<>();
+        NutritionValue output = NutritionCalculator.accumulateNutritionValue(input);
+        NutritionValue expected = new NutritionValue(0, 0, 0, 0);
+        assertEquals(expected.getCalories(), output.getCalories(), Math.ulp(1.0));
```
Comment (olioDuan) [inline NutritionManagement/src/test/NutritionCalculatorTest.java:18]: ` assertEquals`  - good practice to test each field separately as done here, ensuring all aspects of the functionality are verified.

File: NutritionManagement/src/test/NutritionCalculatorTest.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        input.add(new NutritionValue(0.1, 0.3, 0.2, 0.4));
+        input.add(new NutritionValue(0.5, 0.4, 0.3, 0.2));
+        NutritionValue output = NutritionCalculator.accumulateNutritionValue(input);
+        NutritionValue expected = new NutritionValue(0.6, 0.7, 0.5, 0.6);
+        assertEquals(expected.getCalories(), output.getCalories(), Math.ulp(1.0));
+        assertEquals(expected.getProtein(), output.getProtein(), Math.ulp(1.0));
+        assertEquals(expected.getCarbs(), output.getCarbs(), Math.ulp(1.0));
+        assertEquals(expected.getFat(), output.getFat(), Math.ulp(1.0));
+    }
+}
```
Comment (olioDuan) [inline NutritionManagement/src/test/NutritionCalculatorTest.java:36]: Some  suggestions about add more tests for comprehensive coverage _(just a comment)_. 

A test case with:
> a single `NutritionValue` object in the list.
> `null` as input to check how the method handles it.
> a larger list of `NutritionValue` objects to ensure the method scales correctly.

File: NutritionManagement/src/test/NutritionCalculatorTest.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        ArrayList<NutritionValue> input = new ArrayList<>();
+        input.add(new NutritionValue(0.1, 0.3, 0.2, 0.4));
+        input.add(new NutritionValue(0.5, 0.4, 0.3, 0.2));
+        NutritionValue output = NutritionCalculator.accumulateNutritionValue(input);
+        NutritionValue expected = new NutritionValue(0.6, 0.7, 0.5, 0.6);
+        assertEquals(expected.getCalories(), output.getCalories(), Math.ulp(1.0));
+        assertEquals(expected.getProtein(), output.getProtein(), Math.ulp(1.0));
+        assertEquals(expected.getCarbs(), output.getCarbs(), Math.ulp(1.0));
+        assertEquals(expected.getFat(), output.getFat(), Math.ulp(1.0));
+    }
```
Comment (olioDuan) [inline NutritionManagement/src/test/NutritionCalculatorTest.java:35]: Well-implemented with good practices in unit testing. 
The tests are clear, and the scenarios covered are relevant.
Good job @DerekY123

**Review by olioDuan — APPROVED** [review]
**For `NutritionCalculato`r Refactoring:**
Makes the code more concise and readable without sacrificing functionality.

**For new added `NutritionCalculatorTest` Class:**
Tests are clear and maintainable, show good coding practices.

**Pull request merged**